### PR TITLE
Enhancement/price ticker sections

### PR DESCRIPTION
**Pull request by einsteinx2**
**Is this pull request adding a feature or fixing a bug?**  
enhancement

**Does this pull request close an issue? If so, which one?**
#397 

**What does this pull request do? What does it change?**
Adds sections to price ticker to make it more readable

### CODE REVIEW

File: Balance/Shared/View Models/PriceTickerTabViewModel.swift [hunk 1]
```diff
@@ -8,39 +8,85 @@
...
+    
+    var currencies = [[Currency]]()
+    var showPortfolio: Bool {
+        return currencies.count > 0 && currencies[0].count > 0
+    }
     
     func reloadData() {
-        let topCurrencyCodes = ["BTC", "ETH", "LTC", "BCH", "XRP", "DASH", "MIOTA", "ETC", "XMR", "LSK", "STEEM", "GNT", "ZRX"]
+        // Find all currencies owned by the user
+        var portfolioCurrenciesSet = Set<Currency>()
```
Comment (wolffan) [inline Balance/Shared/View Models/PriceTickerTabViewModel.swift]: I see.
This seems likely to go somewhere else (at least it's own method)

File: Balance/Shared/View Models/PriceTickerTabViewModel.swift [hunk 2]
```diff
@@ -8,39 +8,85 @@
...
     }
     
     func numberOfSections() -> Int {
-        return 1
+        return showPortfolio ? 3 : 2
     }
     
     func numberOfRows(inSection section: Int) -> Int {
-        return currencies.count
+        let adjustedSection = showPortfolio ? section : section + 1
```
Comment (wolffan) [inline Balance/Shared/View Models/PriceTickerTabViewModel.swift]: this line should be it's own method probs... lot of repetition

File: Balance/Shared/Data Model/Other Models/Currency.swift [hunk 3]
```diff
@@ -162,9 +189,24 @@ public enum CryptoCurrency: String {
...
+    
+    // Hashable
+    public var hashValue: Int {
+        // Connect alternate BTC symbol XBT
+        if code == "BTC" || code == "XBT" {
+            return "BTC".hashValue
+        }
+        
+        // Connect alternate MIOTA symbols IOT and IOTA
+        if code == "IOTA" || code == "IOT" || code == "MIOTA" {
```
[the remark this answers is not in the document]
Comment (einsteinx2) [reply Balance/Shared/Data Model/Other Models/Currency.swift]: It was half done. I had forgotten to override hashValue as well, so Set collections treated them as separate.

File: Balance/Shared/View Models/PriceTickerTabViewModel.swift (hunk 1 above)
Comment (einsteinx2) [reply Balance/Shared/View Models/PriceTickerTabViewModel.swift]: Yeah I can break these into methods. Sorry old habits die hard :P

File: Balance/Shared/View Models/PriceTickerTabViewModel.swift (hunk 2 above)
Comment (einsteinx2) [reply Balance/Shared/View Models/PriceTickerTabViewModel.swift]: agreed

**Pull request merged**